### ISSUE TITLE
The ability for mapper to pin a comment.

### ISSUE THREAD

**Issue by NevarrenF** [ppy/osu-web#6574]
Basically letting the mapper pin a comment like how Ephemeral did on the Aspire final results.
![chrome_raliqIgI3k](https://user-images.githubusercontent.com/62331647/91665387-ccdbe080-eaba-11ea-9c4c-97078176f685.png)

It would be nice as some people tend to read comments and ignore map descriptions. Sometimes a mapper may want to link alternate mp3s, oszs, or even needs to provide a update on the status of the mapset.

Right now the only real way to kinda pin a comment is to make sure its the most upvoted one which doesn't ensure people will even see it as some will inevitably sort by "Recent" instead of "Top".

![chrome_I4NRCcydTW](https://user-images.githubusercontent.com/62331647/91665393-d82f0c00-eaba-11ea-919f-c60f4a58969c.png)

vs

![chrome_1sG2UGR0cY](https://user-images.githubusercontent.com/62331647/91665395-da916600-eaba-11ea-9684-46af958d905e.png)

**Comment by peppy** [comment]
Is there a reason the details area can't be used for this?

**Comment by Nivalyx** [comment]
i think this is more of an issue of convenience if any. the details/description field of a beatmap as it stands are being positioned in an area where there's a lot of other information going around, and with how it only occupies like a quarter of the screen in a single view this often left things being overlooked by many quite often:

![image](https://user-images.githubusercontent.com/36564236/92332350-8d5b5880-f0a7-11ea-9868-349d7aca7849.png)

the comments section, however, is simply much more convenient to look at with how it is a) occupying a larger space and b) is not being distraught by a flow of data around it. also the fact that the "Read More" button on long comments reveals a lot more with much less effort compared to how scrolling through a beatmap's description would be really contributes to this as well i suppose...

![image](https://user-images.githubusercontent.com/36564236/92332481-6baea100-f0a8-11ea-9d51-c67aa5dcf287.png)

**Comment by NevarrenF** [comment]
That pretty hits the nail on the head, I think at least from experience people just jump to comments and not scroll through.

**Comment by TicClick** [comment]
chimming in to say that, once implemented, this _should_ be allowed for moderators as well

**Comment by peppy** [comment]
I think this should be on request, and not automatically allowed to all mappers, if anything.

Also I have no idea why it should be there for moderators?

**Comment by cl8n** [comment]
I think these examples are misusing comments as a stand-in for the description and it'd be better to resolve issues with visibility and usability of the description area instead. if there is a good reason for wanting a comment pinned though, admin can already do that for you

**Comment by NevarrenF** [comment]
The reason is because people just gloss right over the description where a pinned comment is an extra way to make sure people see things. Also I don't think it would be a good idea to make every person who wants to pin something to go find an admin (I dont know who admins even are past Eph and peppy let alone how to ask them) 

The example I gave actually has the exact same message in the description but people still commented asking for other covers. 

The point I guess I'm trying to make is with the ability to pin comments it's another way to allow more people to see a map announcement for lack of a better term.

**Comment by TicClick** [comment]
> I think this should be on request, and not automatically allowed to all mappers, if anything.
> 
> Also I have no idea why it should be there for moderators? Care to explain the need for that?

sure. something may be not worth adding to the description, but is still necessary for people to consider before posting:

- a comment saying "nowhere, it's disallowed" under maps of songs you received DMCA notices for, in response to all "WHERE IS THE DOWNLOAD LINK???" questions;
- anything that is related to a situation that is rapidly changing, such as the map being disqualified and worked on.

in other words, when something has more relation to what people may comment on rather than the beatmap set itself

**Comment by peppy** [comment]
both of these can be explained in the description, which is mutable.

**Comment by Kagetsu24** [comment]
may i ask what is the reason this is not allowed though? while previous examples didn't provide "strong" reasons as to why this should be added, i think they're still valid. pinning comments on content you made is pretty standard on other platforms such as youtube, and has become a reasonable way of providing more info.
aside from that, i think there are things that the description box cannot replace. a pinned comment feels much more personal in my opinion, because people can reply to it. mappers could have the ability to pin comments from others too, which could eventually lead to people writing more meaningful things on comments. it may be the case that someone wrote a really nice review about your map and you felt like pinning it. i can definitely see something like that happening. admittedly, this could be used for jokes too but i think that it's a pretty normal thing on the internet. also, i think that letting mappers have some fun is harmless at the end of the day.

I don't know if there's fear that this would lead to bad behavior from mappers but maybe add the option to pin a comment only for ranked/loved maps? we have the ability to report comments and moderators so i don't think this would be much of an issue anyway

**Comment by cl8n** [comment]
since my last comment here I talked to a few people who wanted this feature and said similar things as @Kagetsu24 . pinning comments is more of a social feature than anything, which I didn't consider earlier

also, even though I still think the examples in https://github.com/ppy/osu-web/issues/6574#issuecomment-687862756 are more appropriate for the description, a lot of mappers already prefer posting a comment to add more info. having the pin feature available would hopefully improve the way people share things, not make it worse.
